### ISSUE TITLE
HTML builder dump_inventory needs to use the config's encoding for entries

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#426]
I don't know if this is the right fix for it, but I was getting errors with some documents that had German characters in titles and names with Python 2.6 on Mac OS X.  This patch fixed it.

```
# Sphinx version: 1.0pre/c3a4ec6a4e32
# Docutils version: 0.6 release
# Jinja2 version: 2.2.1
Traceback (most recent call last):
  File "/Users/hobu/hg/sphinx/sphinx/cmdline.py", line 173, in main
    app.build(force_all, filenames)
  File "/Users/hobu/hg/sphinx/sphinx/application.py", line 205, in build
    self.builder.build_update()
  File "/Users/hobu/hg/sphinx/sphinx/builders/__init__.py", line 198, in build_update
    'out of date' &#37; len(to_build))
  File "/Users/hobu/hg/sphinx/sphinx/builders/__init__.py", line 254, in build
    self.write(docnames, list(updated_docnames), method)
  File "/Users/hobu/hg/sphinx/sphinx/builders/__init__.py", line 293, in write
    self.write_doc(docname, doctree)
  File "/Users/hobu/hg/sphinx/sphinx/builders/html.py", line 407, in write_doc
    self.docwriter.write(doctree, destination)
  File "/Library/Python/2.6/site-packages/docutils-0.6-py2.6.egg/docutils/writers/__init__.py", line 76, in write
    self.translate()
  File "/Users/hobu/hg/sphinx/sphinx/writers/html.py", line 37, in translate
    self.document.walkabout(visitor)
  File "/Library/Python/2.6/site-packages/docutils-0.6-py2.6.egg/docutils/nodes.py", line 173, in walkabout
    if child.walkabout(visitor):
  File "/Library/Python/2.6/site-packages/docutils-0.6-py2.6.egg/docutils/nodes.py", line 173, in walkabout
    if child.walkabout(visitor):
  File "/Library/Python/2.6/site-packages/docutils-0.6-py2.6.egg/docutils/nodes.py", line 186, in walkabout
    visitor.dispatch_departure(self)
  File "/Library/Python/2.6/site-packages/docutils-0.6-py2.6.egg/docutils/nodes.py", line 1617, in dispatch_departure
    return method(node)
  File "/Users/hobu/hg/sphinx/sphinx/writers/html.py", line 482, in depart_title
    _('Permalink to this headline'))
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 20: ordinal not in range(128)





diff -r f8acb3ce3fd3 sphinx/builders/html.py
--- a/sphinx/builders/html.py   Wed May 26 00:25:41 2010 +0200
+++ b/sphinx/builders/html.py   Wed May 26 10:51:32 2010 -0500
@@ -747,7 +747,12 @@
                     if anchor.endswith(name):
                         # this can shorten the inventory by as much as 25&#37;
                         anchor = anchor[:-len(name)] + '$'
-                    uri = self.get_target_uri(docname) + '#' + anchor
+                    uri = self.get_target_uri(docname) + '#' + anchor 
+                    encoding = self.config.source_encoding
+                    uri = uri.encode(encoding)
+                    dispname = dispname.encode(encoding)
+                    name = name.encode(encoding)
+                    domainname = domainname.encode(encoding)
                     if dispname == name:
                         dispname = '-'
                     f.write(compressor.compress(
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/426
- Originally reported by: Anonymous
- Originally created at: 2010-05-26T17:54:20.793

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2010-05-26 17:51:44+00:00_

Thanks for the report, should be fixed in <<cset 0d4c3a31b63e58602e469e65a164ddcfc1e4443e>>.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2010-05-30 18:07:22+00:00_

Removing version: 1.0b1 (automated comment)
